src/deploy-cromwell-on-azure/KubernetesManager.cs

```diff
@@ -0,0 +1,261 @@
+            }
+
+            if (!string.IsNullOrWhiteSpace(configuration.CustomCromwellImagePath))
+            {
+                values.Images["cromwell"] = configuration.CustomCromwellImagePath;
+            }
+
+            var writer = new StreamWriter(Path.Join("scripts", "helm", "values.yaml"));
+            writer.Write(Yaml.SaveToString(values));
+            writer.Close();
+        }
+
+        private async Task ExecHelmProcess(string command)
+        {
+            var p = new Process();
+            p.StartInfo.UseShellExecute = false;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.FileName = configuration.HelmExePath;
+            p.StartInfo.Arguments = command;
+            p.Start();
+
+            if (configuration.DebugLogging)
+            {
+                var line = p.StandardOutput.ReadLine();
+                while (line != null)
+                {
+                    ConsoleEx.WriteLine("HELM: " + line);
+                    line = p.StandardOutput.ReadLine();
+                }
+            }
+
+            await p.WaitForExitAsync();
+        }
+
+        public async Task ExecuteCommandsOnPod(IKubernetes client, string podName, IEnumerable<string[]> commands, TimeSpan timeout)
+        {
+            var printHandler = new ExecAsyncCallback(async (stdIn, stdOut, stdError) =>
+            {
+                using (var reader = new StreamReader(stdOut))
+                {
+                    var line = await reader.ReadLineAsync();
+
+                    while (line != null)
+                    {
+                        if (configuration.DebugLogging)
+                        {
+                            ConsoleEx.WriteLine(podName + ": " + line);
+                        }
+                        line = await reader.ReadLineAsync();
+                    }
+                }
+
+                using (var reader = new StreamReader(stdError))
+                {
+                    var line = await reader.ReadLineAsync();
+
+                    while (line != null)
+                    {
+                        if (configuration.DebugLogging)
+                        {
+                            ConsoleEx.WriteLine(podName + ": " + line);
+                        }
+                        line = await reader.ReadLineAsync();
+                    }
+                }
+            });
+
+            var pods = await client.ListNamespacedPodAsync(configuration.AksCoANamespace);
+            var workloadPod = pods.Items.Where(x => x.Metadata.Name.Contains(podName)).FirstOrDefault();
+
+            if (!await WaitForWorkloadWithTimeout(client, podName, timeout, cts.Token))
+            {
+                throw new Exception($"Timed out waiting for {podName} to start.");
+            }
+            // For some reason even if a pod says it ready, calls made immediately will fail. Wait for 40 seconds for safety. 
```

Comment:
Is there a way to ping the pod instead of waiting for arbitrary amount of time?

Reply:
Not exactly, pinging the pod that's not running will result in the same websocket exception that you get if you try to run a command on the pod. I rewrote it to retry execution every 5 secs with a 40 sec timeout rather than waiting for 40 secs though.

Reply:
Fixed with Polly.